### PR TITLE
mtev_cluster PR-3

### PR DESCRIPTION
**Pull request by HeinrichHartmann**
See commit messages

### CODE REVIEW

File: src/mtev_capabilities_listener.c [hunk 1]
```diff
@@ -83,13 +83,11 @@ mtev_capabilities_listener_init(void) {
...
 
-void
+int
 mtev_capabilities_add_feature(const char *feature, const char *version) {
   feature = strdup(feature);
   if(version) version = strdup(version);
-  if(!mtev_hash_store(&features, feature, strlen(feature), (void *)version))
-    mtevL(mtev_error, "Feature conflict! %s version %s\n",
-          feature, version ? version : "unpecified");
+  return mtev_hash_set(&features, feature, strlen(feature), (void *)version, NULL, NULL);
```
Comment (postwait) [inline src/mtev_capabilities_listener.c]: This is supposed to fail.

If you want to replace a feature, we need to add a new API.  mtev_capabilities_replace_feature.

Or... it doesn't fit as a feature and we should just query the cluster API.

Comment (HeinrichHartmann) [reply src/mtev_capabilities_listener.c]: yeah. I could not make this work without changes to mtev_capabilities_listener.
Changing `mtev_capabilities_add_feature` seemed less intrusive than creating an entirely new function.

The choice is up to you. I can add a "mtev_capabilities_add_or_replace_feature" function or leave it out entirely. No strong opinion here.

**Comment by postwait** [comment]
Let's leave it out for now

**Comment by HeinrichHartmann** [comment]
OK. _removed last commit_.

**Pull request merged**